### DISCUSSION TITLE
loop over nodal variable values of current element in computeQpProperties()

### DISCUSSION THREAD

**Discussion by bylore** [idaholab/moose#28613] [Q&A General] [answered]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

hi there, i have a material property whose local value need to be modified according to the surrounding nodal variable value. In one element, before i determine the value of this material property at the quadpoint, i should loop over all the node of the current element and check if the nodal variable value meet the requirement. But i don't know how to implement this, could you give me some advice? Thanks!

**Comment by GiudGiud** [comment, marked as answer]
Hello

From the current element attribute, you can get the list of nodes.
Then you can use either getNodalValue or the NodalArg (see MooseFunctor.h) to get the nodal values of the variable.

Guillaume

**Reply by bylore** [reply]
It works. Thank you！
